hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -100,42 +111,89 @@ public void run() {
 
   public void triggerContainerHealthCheck() {
     lock.writeLock().lock();
+    Map<String, Map<String, Long>> unhealthyContainerStateStatsMap;
     try {
+      unhealthyContainerStateStatsMap = new HashMap<>(Collections.emptyMap());
+      initializeUnhealthyContainerStateStatsMap(
+          unhealthyContainerStateStatsMap);
       long start = Time.monotonicNow();
       long currentTime = System.currentTimeMillis();
-      long existingCount = processExistingDBRecords(currentTime);
+      long existingCount = processExistingDBRecords(currentTime,
+          unhealthyContainerStateStatsMap);
       LOG.info("Container Health task thread took {} milliseconds to" +
               " process {} existing database records.",
           Time.monotonicNow() - start, existingCount);
       start = Time.monotonicNow();
       final List<ContainerInfo> containers = containerManager.getContainers();
       containers.stream()
           .filter(c -> !processedContainers.contains(c))
-          .forEach(c -> processContainer(c, currentTime));
+          .forEach(c -> processContainer(c, currentTime,
+              unhealthyContainerStateStatsMap));
```

Comment:
Unrelated to this PR but this `containerManager.getContainers()` doesn't seem to be using pagination right now. Should this pose any memory issue (e.g. OOM) we should improve this.

File a jira for this?

Reply:
Yes, good catch, Filed a separate JIRA [HDDS-9819](https://issues.apache.org/jira/browse/HDDS-9819) for handling this.